### PR TITLE
Remove ?namespace= param handling/defaulting

### PR DESCRIPTION
**Pull request by liggitt**
Removes obsolete handling of namespace parameter and defaulting behavior

### CODE REVIEW

File: pkg/apiserver/api_installer.go [hunk 1]
```diff
@@ -730,6 +695,9 @@ func (n scopeNaming) GenerateLink(req *restful.Request, obj runtime.Object) (pat
 			return "", "", err
 		}
 	}
+	if len(name) == 0 {
+		return "", "", errEmptyName
+	}
```
Comment (liggitt) [inline pkg/apiserver/api_installer.go]: @smarterclayton can you take a look at this? `rootScopeNaming#GenerateLink` returned an err if Name was empty, and `TestSelfLinkSkipsEmptyName` was complaining unless scopeNaming#GenerateLink behaved similarly

**Comment by smarterclayton** [comment]
LGTM

**Comment by fabioy** [comment]
@k8s-bot ok to test

I guess Jenkins got stuck. Trying again for green.

**Comment by saad-ali** [comment]
Hey guys, thanks for the pull request -- we sincerely appreciate the work that goes into sending us a change.

As you might know, we're currently in a feature-freeze and general code slush as we work toward our 1.0 release. We're currently only accepting changes that meet any of the following criteria:
  1. Changes directly related to v1.0 issues: [link](https://github.com/GoogleCloudPlatform/kubernetes/issues?q=is%3Aopen+is%3Aissue+milestone%3Av1.0)
  2. Changes addressing P0 bugs: [link](https://github.com/GoogleCloudPlatform/kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3Apriority%2FP0)
  3. Changes to supported platforms to bring them up to par with the primary development platforms
  4. Docs, user experience, and test related fixes

Our initial triage indicates that this is a non-critical change at this time, so we'd like to leave it open and revisit it after 1.0.  We expect the freeze to wrap up in early to mid July.

If you feel that this triage is incorrect, please respond and let us know why.  

Thanks again for your contribution!  The community is really what has made this project so exciting.

CC @bgrant0607, @brendanburns

**Comment by bgrant0607** [comment]
Part of #8087

File: pkg/apiserver/api_installer.go [hunk 2]
```diff
@@ -282,7 +281,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		actions = appendIf(actions, action{"CONNECT", itemPath + "/{path:*}", nameParams, namer}, isConnecter && connectSubpath)
 
 	} else {
-		// v1beta3 format with namespace in path
+		// v1beta3+ format with namespace in path
 		if scope.ParamPath() {
```
Comment (bgrant0607) [inline pkg/apiserver/api_installer.go]: Why not just completely remove ParamPath()?

**Pull request merged**